src/libraries/HooksLib.sol

```diff
@@ -1,25 +1,34 @@
-// SPDX-License-Identifier: UNLICENSED
+// SPDX-License-Identifier: GPL-2.0-or-later
 pragma solidity ^0.8.23;
 
 import {IHook} from "../interfaces/IHook.sol";
 import {IValidationHook} from "../interfaces/IValidationHook.sol";
 import {IExecutionHook} from "../interfaces/IExecutionHook.sol";
 import {PackedUserOperation} from "account-abstraction/interfaces/PackedUserOperation.sol";
 
+/// @title HooksLib
+/// @notice Hooks are invoked by inspecting the least significant bits of the address it is deployed to
+/// For example, a hook deployed to address: 0x000000000000000000000000000000000000000a
+/// has the lowest bits '0001 1010' which would cause the 'VALIDATE_USER_OP', 'BEFORE_EXECUTE', and 'AFTER_EXECUTE' hooks to be used.
 /// @author Inspired by https://github.com/Uniswap/v4-core/blob/main/src/libraries/Hooks.sol
 library HooksLib {
-    uint160 internal constant VERIFY_SIGNATURE_FLAG = 1 << 0;
-    uint160 internal constant VALIDATE_USER_OP_FLAG = 1 << 1;
-    uint160 internal constant IS_VALID_SIGNATURE_FLAG = 1 << 2;
+    /// @notice Internal constant hook flags
+    uint160 internal constant AFTER_VERIFY_SIGNATURE_FLAG = 1 << 0;
+    uint160 internal constant AFTER_VALIDATE_USER_OP_FLAG = 1 << 1;
+    uint160 internal constant AFTER_IS_VALID_SIGNATURE_FLAG = 1 << 2;
     uint160 internal constant BEFORE_EXECUTE_FLAG = 1 << 3;
     uint160 internal constant AFTER_EXECUTE_FLAG = 1 << 4;
 
+    /// @notice Hook did not return its selector
     error InvalidHookResponse();
 
+    /// @notice Returns whether the flag is configured for the hook
     function hasPermission(IHook self, uint160 flag) internal pure returns (bool) {
         return uint160(address(self)) & flag != 0;
     }
 
+    /// @notice Handles the afterValidateUserOp hook
+    /// @return validationData encoded according to ERC-4337 spec
```

Comment:
same add something about how it can override internal validationData

Reply:
huh I just thought of something.. If the hook doesn't want to override the validationData it has to figure out how to construct the validationData itself, maybe if it returns some sentinel value, it should not default override?

Reply:
Hmm interesting, I guess it doesn't seem very difficult for the hook to return validation data? But I agree the default override behavior is a little interesting

Reply:
ya its not necessary but just a callout